static/src/js/util/map/getQueryFromShapefileFeature.ts

```diff
@@ -16,19 +33,65 @@ const getQueryFromShapefileFeature = (feature: Feature) => {
   // Shapefiles can have altitude in their coordinates. This shows up as a 3rd value in each coordinate pair (e.g., [longitude, latitude, altitude]).
   // For CMR spatial queries, we only need the longitude and latitude values.
   let geographicCoordinatesWithoutAltitude = geographicCoordinates
+
+  // Check for altitude in coordinates - handle different geometry structures
+  const hasAltitude = () => {
+    if (!geographicCoordinates || !geographicCoordinates[0]) return false
+
+    // For MultiPolygon
+    if (Array.isArray(geographicCoordinates[0][0][0])) {
+      return geographicCoordinates[0][0][0].length === 3
+    }
+
+    // For Polygon, LineString
+    if (Array.isArray(geographicCoordinates[0][0])) {
+      return geographicCoordinates[0][0].length === 3
+    }
+
+    // For Point, MultiPoint
+    return geographicCoordinates[0].length === 3
+  }
+
+  // Only run altitude detection for geometry types that use geographicCoordinates
   if (
-    geographicCoordinates
-    && geographicCoordinates[0] // Array of coordinates
-    && geographicCoordinates[0][0] // First coordinate
-    && geographicCoordinates[0][0].length === 3 // If the coordinate has 3 values (longitude, latitude, altitude)
+    geometryType !== spatialTypes.CIRCLE
+    && geometryType !== spatialTypes.POINT
+    && hasAltitude()
   ) {
-    geographicCoordinatesWithoutAltitude = geographicCoordinates.map(
-      (coords: number[][]) => coords.map(
-        (coord: number[]) => coord.slice(0, 2)
-      )
-    )
+    geographicCoordinatesWithoutAltitude = removeAltitudeFromCoordinates(geographicCoordinates)
+  }
+
+  // Transformation functions for different geometry types
+  const transformPolygonToString = (polygon: number[][][]): string => polygon[0].flat().join(',')
```

Comment:
A few things:

1. You can use the new types for the inputs from these functions
2: Why are we pulling only the first polygon on line 65? Is it possible that there are multiple polygons?
3: I'd like to see these functions commented with JSDoc style comments. I also feel like they might make sense as standalone utils with tests rather than existing in this file.
4: Lastly, I'm tempted not to stress it right now but I wonder if this could be cleaned up with a switch statement or something. Not a huge fan of the repetitive `if` statements when theres seems to be a fairly concrete pattern in each.